src/plugins/issue-archiver/index.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+
+const createScheduler = require("probot-scheduler");
+const moment = require("moment");
+
+const SEARCH_INTERVAL_MS = 6 * 60 * 60 * 1000; // 6 hours
+const ARCHIVAL_AGE_DAYS = 180;
+const ARCHIVED_LABEL = "archived due to age";
+
+/**
+ * Checks whether the current repository has a `ARCHIVED_LABEL` label
+ * @param {probot.Context} context Probot context for the current repository
+ * @returns {Promise<boolean>} `true` if the repository has a label with appropriate name
+ */
+async function hasArchivedLabel(context) {
+    const allLabels = await context.github.paginate(
+        context.github.issues.getLabels(context.repo()),
+        res => res.data
+    );
+
+    return allLabels.some(label => label.name === ARCHIVED_LABEL);
+}
+
+
+/**
+ * Creates a search query to look for issuesin a given repo which have been closed for at least
+ * `ARCHIVAL_AGE_DAYS` days since the current date, and have not yet been archived.
+ * @param {string} options.owner The owner of the repo where issues should be searched
+ * @param {string} options.repo The name of the repo where issues should be searched
+ * @returns {string} A search query to send to the GitHub API
+ */
+function createSearchQuery({ owner, repo }) {
+    const earliestDayWithinArchivalAge = moment().subtract({ days: ARCHIVAL_AGE_DAYS });
+
+    return [
+        "is:closed",
+        `repo:${owner}/${repo}`,
+        `-label:"${ARCHIVED_LABEL}"`,
```

Comment:
Maybe we can use `-is:locked`?

Reply:
Oh, that's a good idea. I hadn't realized `is:locked` is supported (it seems to only work when using the API, but not when using regular filters like [this](https://github.com/eslint/eslint/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+is%3Alocked)).

Reply:
Actually, https://api.github.com/search/issues?q=repo%3Aeslint%2Feslint%20-is%3Alocked seems to return a 500 error. It seems like `is:locked` works, but `-is:locked` doesn't work.

Reply:
Oh... I see. :cry:

Reply:
(I found it a little while ago, I type it in https://github.com/eslint/eslint/search?utf8=%E2%9C%93&q=is%3Alocked&type=Issues)

Reply:
I added https://github.com/eslint/eslint-github-bot/commit/224d1c7af6d41912119d6c6a5b2bfae33a40eeab to filter out locked issues manually.